packages/fast-tooling-react/src/css-editor/box-shadow/box-shadow.tsx

```diff
@@ -71,112 +82,95 @@ export default class CSSBoxShadow extends Foundation<
+            cssBoxShadow_colorInputRegion,
+            cssBoxShadow_opacityInput,
+            cssBoxShadow_controlRegion,
+            cssBoxShadow_label,
+            cssBoxShadow_xInput,
+            cssBoxShadow_yInput,
+            cssBoxShadow_blurInput,
+        }: Partial<CSSBoxShadowClassNameContract> = this.props.managedClasses;
+
         return (
-            <div className={get(this.props, "managedClasses.cssBoxShadow")}>
-                <div
-                    className={get(this.props, "managedClasses.cssBoxShadow_colorRegion")}
-                >
-                    <label
-                        className={get(this.props, "managedClasses.cssBoxShadow_label")}
-                    >
-                        Shadow
-                    </label>
+            <div className={this.generateClassNames()}>
+                <div className={cssBoxShadow_control}>
                     <div
-                        className={get(this.props, "managedClasses.cssBoxShadow_control")}
+                        className={cssBoxShadow_colorInputRegion}
+                        style={{ background: this.state.boxShadowColor }}
                     >
-                        <div
-                            className={get(
-                                this.props,
-                                "managedClasses.cssBoxShadow_colorInputRegion"
-                            )}
-                            style={{ background: this.state.boxShadowColor }}
-                        >
-                            <input
-                                type={"color"}
-                                style={{ opacity: 0, width: "100%" }}
-                                value={this.state.boxShadowColor}
-                                onChange={this.handleBoxShadowColorOnChange}
-                            />
-                        </div>
                         <input
-                            className={get(
-                                this.props,
-                                "managedClasses.cssBoxShadow_opacityInput"
-                            )}
-                            type={"number"}
-                            min={0}
-                            max={100}
-                            step={1}
-                            placeholder={"50"}
-                            value={Math.round(this.state.boxShadowOpacity * 100)}
-                            onChange={this.handleBoxShadowOpacityOnChange}
+                            type={"color"}
+                            style={{ opacity: 0, width: "100%" }}
+                            value={this.state.boxShadowColor}
+                            onChange={this.handleBoxShadowColorOnChange}
+                            disabled={this.props.disabled}
+                            onFocus={this.props.reportValidity}
+                            onBlur={this.props.updateValidity}
+                            ref={this.props.elementRef as React.Ref<HTMLInputElement>}
```

Comment:
what is the type of elementRef? Can we do this without coercing the type?

Reply:
type of elementRef is `React.Ref<FormHTMLElement>` If I don't coerce the type I get an error stating: `Type 'Ref<FormHTMLElement>' is not assignable to type 'LegacyRef<HTMLInputElement>'`